server/src/main/java/org/elasticsearch/index/fielddata/MultiGeoValues.java

```diff
@@ -166,28 +167,28 @@ public static GeoShapeValue missing(String missing) {
         public final double posLeft;
         public final double posRight;
 
-        BoundingBox(Extent extent) {
-            this.top = GeoEncodingUtils.decodeLatitude(extent.top);
-            this.bottom = GeoEncodingUtils.decodeLatitude(extent.bottom);
+        BoundingBox(Extent extent, CoordinateEncoder coordinateEncoder) {
```

Comment:
I wonder if it would simplify things if we "taught" extent how to decode itself, by passing it as a constrictor parameter into Extent?

Reply:
that is a good point! I will explore this when it is being used more in intersection queries in CellIdSource